l3kernel/testfiles/m3str002.lvt

```diff
@@ -56,5 +56,26 @@
     \tl_if_eq:NNTF \l_tmpa_tl \l_tmpb_tl \TRUE \FALSE
   }
 
+\OMIT
+\tl_set:Nn \l_tmpa_tl { a }
+\tl_set:Nn \l_tmpb_tl { a }
+\TIMO
+
+\TESTEXP { MD5~hash }
+  {
+    \str_mdfive_hash:n { foo }
+    \NEWLINE
+    \str_mdfive_hash:n { & \par ! ## }
+    \NEWLINE
+    \str_mdfive_hash:n { \l_tmpa_tl }
+    \NEWLINE
+    \str_mdfive_hash:n { \l_tmpb_tl }
+    \NEWLINE
+    \str_mdfive_hash:e { \l_tmpa_tl }
+    \NEWLINE
+    \str_mdfive_hash:e { \l_tmpb_tl }
```

Comment:
Just self-talk, not a review.  This is certainly not the first time I learnt that some primitives fully expand their args before taking them as string.  I had to use sth like
```tex
% md5 for string "a"
\str_if_eq:eeTF
  { \str_mdfive_hash:e { \l_tmpa_tl } }
  { \str_mdfive_hash:e { a } }
  { EQUAL } { INEQUAL }
\NEWLINE
% md5 for string "\l_tmpa_tl "
\str_if_eq:eeTF
  { \str_mdfive_hash:n { \l_tmpa_tl } }
  { \str_mdfive_hash:e { \c_backslash_str l_tmpa_tl ~ } }
  { EQUAL } { INEQUAL }
```
to convince myself what `\tex_mdfivesum:D` does.